apps/ios/LiveVideo/LiveVideo/Views/Participants/ViewersViewModel.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  Copyright (C) 2021 Twilio, Inc.
+//
+
+import Combine
+
+class ViewersViewModel: ObservableObject {
+    @Published var viewersWithRaisedHand: [SyncUsersStore.User] = []
+    @Published var viewersWithoutRaisedHand: [SyncUsersStore.User] = []
```

Comment:
The UX design calls for displaying all the viewers in a list with the raised hands at the top.

Long story short we must use separate sync maps for viewers and raised hands because of the way sync works.

The viewers map contains all viewers including viewers that may have their hand raised. And so the client must filter users out of the viewers map if they have their hand raised. Why not have the server do this filtering?

1. It seemed more general purpose and UI agnostic to simply have a set of all viewers and then the client can do whatever they want with it.
1. If the server did the filtering the client would not be able to animate list updates in the UI because the updates to the viewers map and raised hands map would come in at different times. And even worse we would see strange things like a user showing up twice or not at all momentarily. In my next ticket we will see issues like this when the user switches between viewers and speakers but I think that won't be bad because the user is switching between different list sections in the UI as opposed to changing position inside one section. It seems like the latter would look more like a bug if not done well compared to the former. Or is it just me?

Make sense? Thoughts?

Reply:
I'm feeling good about this and will resolve.

Reply:
This approach is reasonable to me. A raised hand user __is a__ viewer so this is accurate IMO.